project/CoursierWorkflow.scala

```diff
@@ -0,0 +1,85 @@
+import sbt._
+import sbtghactions.GenerativePlugin
+import sbtghactions.GenerativePlugin.autoImport._
+
+object CoursierWorkflow extends AutoPlugin {
+  override def requires: Plugins = GenerativePlugin
+
+  object autoImport {
+    val coursierSetup = settingKey[WorkflowStep]("coursier setup step")
+    val coursierCache = settingKey[WorkflowStep]("coursier cache step")
+
+    def Sbtn(
+      commands: List[String],
+      id: Option[String] = None,
+      name: Option[String] = None,
+      cond: Option[String] = None,
+      env: Map[String, String] = Map()
+    ): WorkflowStep = {
+
+      def indent(output: String, level: Int): String = {
+        val space = (0 until level * 2).map(_ => ' ').mkString
+        (space + output.replace("\n", s"\n$space")).replaceAll("""\n[ ]+\n""", "\n\n")
+      }
+
+      def isSafeString(str: String): Boolean =
+        !(str.indexOf(':') >= 0 || // pretend colon is illegal everywhere for simplicity
+          str.indexOf('#') >= 0 || // same for comment
+          str.indexOf('!') == 0 ||
+          str.indexOf('*') == 0 ||
+          str.indexOf('-') == 0 ||
+          str.indexOf('?') == 0 ||
+          str.indexOf('{') == 0 ||
+          str.indexOf('}') == 0 ||
+          str.indexOf('[') == 0 ||
+          str.indexOf(']') == 0 ||
+          str.indexOf(',') == 0 ||
+          str.indexOf('|') == 0 ||
+          str.indexOf('>') == 0 ||
+          str.indexOf('@') == 0 ||
+          str.indexOf('`') == 0 ||
+          str.indexOf('"') == 0 ||
+          str.indexOf('\'') == 0 ||
+          str.indexOf('&') == 0)
+
+      def wrap(str: String): String =
```

Comment:
This part of the code for wrapping and escaping looks kind of brittle.. Is it really necessary?